parts/linux/cloud-init/artifacts/localdns.sh

```
@@ -0,0 +1,242 @@
+# Replace all occurrences of VnetDNS_Server_IP in localdns corefile.
+sed -i -e "s|VnetDNS_Server_IP|${UPSTREAM_VNET_DNS_SERVERS}|g" "${LOCALDNS_CORE_FILE}" || { echo "Error: updating corefile failed"; exit 1; }
+cat "${LOCALDNS_CORE_FILE}"
+
+# Iptables: build rules.
+# --------------------------------------------------------------------------------------------------------------------
+# These rules skip conntrack for DNS traffic to the local DNS service IPs to save conntrack table space. 
+# OUTPUT rules affect node services and hostNetwork: true pods.
+# PREROUTING rules affect traffic from regular pods.
+IPTABLES='iptables -w -t raw -m comment --comment "Local DNS: skip conntrack"'
+IPTABLES_RULES=()
+for CHAIN in OUTPUT PREROUTING; do
+for IP in ${LOCALDNS_NODE_LISTENER_IP} ${LOCALDNS_CLUSTER_LISTENER_IP}; do
+for PROTO in tcp udp; do
+    IPTABLES_RULES+=("${CHAIN} -p ${PROTO} -d ${IP} --dport 53 -j NOTRACK")
+done; done; done
+
+# Information variables.
+# --------------------------------------------------------------------------------------------------------------------
+DEFAULT_ROUTE_INTERFACE="$(ip -j route get 168.63.129.16 | jq -r '.[0].dev')"
+NETWORK_FILE="$(networkctl --json=short status "${DEFAULT_ROUTE_INTERFACE}" | jq -r '.NetworkFile')"
+NETWORK_DROPIN_DIR="${NETWORK_FILE}.d"
+NETWORK_DROPIN_FILE="${NETWORK_DROPIN_DIR}/70-localdns.conf"
+
+# Cleanup function will be run on script exit/crash to revert config.
+# --------------------------------------------------------------------------------------------------------------------
+function cleanup {
+    # Disable error handling so that we don't get into a recursive loop.
+    set +e
+
+    # Remove iptables rules to stop forwarding DNS traffic.
+    for RULE in "${IPTABLES_RULES[@]}"; do
+        if eval "${IPTABLES}" -C "${RULE}" 2>/dev/null; then
+            eval "${IPTABLES}" -D "${RULE}"
```

can this command fail?

---

or it will just fail silently?

---

Added the checks in the code.